test_linked_list.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+
+def test_node_class():
+    """Test Node."""
+    from linked_list import Node
+    new_node = Node('caterpillar')
+    assert new_node.val == 'caterpillar'
+
+
+def test_search():
+    """Search for node in list and return node."""
+    from linked_list import LinkedList
+    fake_list = LinkedList()
+    fake_list.insert('that')
+    assert fake_list.search('that') == 'that'
+
+
+def test_size():
+    """Test if size of list is returned."""
+    from linked_list import LinkedList
+    my_list = LinkedList()
+    my_list.insert('something')
+    my_list.insert('something else')
+    my_list.insert('Oh what the hell, one more thing.')
+    assert my_list.size() == 3
+
+
+def test_display():
+    """Print the list."""
+    from linked_list import LinkedList
+    my_list = LinkedList()
+    my_list.insert(1)
+    my_list.insert(2)
+    assert my_list.size() == len(my_list.display())
+
+
+def test_pop():
+    """Assert first value was removed and returned."""
+    from linked_list import LinkedList
+    my_list = LinkedList()
+    my_list.insert('data')
+    my_list.insert('more data')
+    assert my_list.pop() == 'more data'
+
```

Comment:
pop() is not fully tested here. What about when the list is empty? Does your pop() function do you what you expect in that case? What if you pop() twice in this instance? You should test this more thoroughly, because your pop() function is actually broken, and more testing would prove  it.